restcomm/restcomm.interpreter/src/test/java/org/restcomm/connect/interpreter/mediagroup/StopMediaGroupTest.java

```diff
@@ -0,0 +1,395 @@
+
+            @Override
+            public Actor create() throws Exception {
+                return new VoiceInterpreter(builder.build()){
+
+                    @Override
+                    protected ActorRef downloader() {
+                        return observer;
+                    }
+
+                    @Override
+                    protected ActorRef cache(String path, String uri) {
+                        return downloader;
+                    }
+
+                    @Override
+                    public ActorRef getCache() {
+                        return downloader;
+                    }
+
+                    @Override
+                    protected URI resolve(URI uri){
+                        return uri;
+                    }
+
+                    @Override
+                    protected boolean is(State state) {
+                        //bypass fsm state so the test logic can be reduced
+                        String id = state.getId();
+                        for(int i = 0; i < fsmStateBypass.length; i++){
+                            if(id.equals(fsmStateBypass[i])){
+                                return true;
+                            }
+                        }
+                        //if there is no state to bypass, stick to formal verification
+                        return super.is(state);
+                    }
+                };
+            }
+        });
+        return TestActorRef.create(system, props, "VoiceInterpreter" + System.currentTimeMillis());
+    }
+
+    @Test
+    public void testForkingBusyStop() throws Exception {
+        new JavaTestKit(system){
+            {
+                //creating and starting vi
+                final ActorRef observer = getRef();
+                final ActorRef interpreter = createVoiceInterpreter(observer, "forking");
+                interpreter.tell(new StartInterpreter(observer), observer);
+                expectMsgClass(GetCallInfo.class);
+
+                //notifying vi that the last/only call answered with busy
+                final CallStateChanged callStateChanged = new CallStateChanged(CallStateChanged.State.BUSY);
+                interpreter.tell(callStateChanged, null);
+
+                //check if StopMediaGroup was requested to media actors
+                //this request originally continues through the actors Call > CallController > MediaGroup
+                //reaching media server and stopping whats currently playing. But at this test, both call and
+                //vi actors are using the same observer, so we receive StopMediaGroup directly
+                expectMsgClass(StopMediaGroup.class);
+
+                //emulate the response that comes from media server, through media group, reaching vi
+                interpreter.tell(new MediaGroupResponse<String>("stopped"), observer);
+
+                //check if vi stays still and DO NOT ask for next verb
+                expectNoMsg();
```

Comment:
i like this test cases. i find this expectNoMsg a bit opened to changes, or different behaviors, could we add assertions on current state of Actor using "is(State)" method?

Reply:
I guess that is not possible given the fact method "is(State)" is not part of the akka actor scope, so no way to invoke it from the test scenario. I'll check alternatives, i agree that checking that also would bring more credibility to the results.

Reply:
have you tried using TypedActors,instead of untyped?... for test purposes  think is valid and it shoudl only affect on the code that instantiate the actor in the test code, not the actor code itself...

Reply:
I tried to understand how it works @jaimecasero but couldn't make it. I know VI inherits from untyped, but no ideas on how to fit it in the use case avoiding structural changes. (Does it worth the time/effort to only double check FMS state?)